src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -279,9 +279,30 @@ public AzureAppConfigurationOptions Connect(Uri endpoint, TokenCredential creden
                 throw new ArgumentNullException(nameof(credential));
             }
 
+            return Connect(new List<Uri>() { endpoint }, credential);
+        }
+
+        /// <summary>
+        /// Connect the provider to Azure App Configuration and their replicas using list of endpoints and token credential.
+        /// </summary>
+        /// <param name="endpoints">The list of endpoints of an Azure App Configuration store and its replicas to connect to.</param>
+        /// <param name="credential">Token credential to use to connect.</param>
+        public AzureAppConfigurationOptions Connect(IEnumerable<Uri> endpoints, TokenCredential credential)
+        {
+            if (endpoints == null || !endpoints.Any())
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            if (endpoints.Distinct().Count() != endpoints.Count())
```

Comment:
Does `Distinct()` consider casing?

Reply:
Thanks for pointing out, we actually want a custom comparer because, `endpoints` is actually a an enumerable of `Uri`.
`Distinct()` would by default use the default equality comparer. For our use case, we only care about certain parts of the `Uri` to be equals. (i.e. `Host` and `Port`). So I created a new `EqualityComparer` that would do the comparison as we would like.